rpa-adapter/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/BaseMapper.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.cmc.rpa.mapper;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class BaseMapper {
+    protected <G, S extends G> String extractFromSubclass(G instance, Class<S> subclass, Function<S, String> mapper) {
```

Comment:
prefer to have these as static utility methods, we don't need a base class contract.